collector/eventhub/src/main/java/zipkin/collector/eventhub/ZipkinEventProcessor.java

```diff
@@ -17,63 +17,76 @@
 import com.microsoft.azure.eventprocessorhost.CloseReason;
 import com.microsoft.azure.eventprocessorhost.IEventProcessor;
 import com.microsoft.azure.eventprocessorhost.PartitionContext;
-import zipkin.Codec;
-import zipkin.collector.Collector;
-
-import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import zipkin.Codec;
+import zipkin.Span;
+import zipkin.collector.Collector;
 
 import static zipkin.storage.Callback.NOOP;
 
-public class ZipkinEventProcessor implements IEventProcessor {
+// TODO: we assume this is a single-threaded processor. If not, batching logic must be changed
+final class ZipkinEventProcessor implements IEventProcessor {
+  final Logger logger;
+  final Collector collector;
+  final int checkpointBatchSize;
 
-  private static final Logger logger = Logger.getLogger(ZipkinEventProcessor.class.getName());
+  int countSinceCheckpoint = 0;
 
-  int checkpointBatchingCount = 0;
-  EventHubCollector.Builder config;
-  Collector collector;
+  ZipkinEventProcessor(Collector collector, int checkpointBatchSize) {
+    this(Logger.getLogger(ZipkinEventProcessor.class.getName()), collector, checkpointBatchSize);
+  }
 
-  public ZipkinEventProcessor(EventHubCollector.Builder builder) {
-    config = builder;
-    collector = config.delegate.build();
+  ZipkinEventProcessor(Logger logger, Collector collector, int checkpointBatchSize) {
+    this.logger = logger;
+    this.collector = collector;
+    this.checkpointBatchSize = checkpointBatchSize;
   }
 
   @Override
-  public void onOpen(PartitionContext context) throws Exception {
-    logger.log(Level.INFO,"Opened " + context.getConsumerGroupName());
+  public void onOpen(PartitionContext context) {
+    logger.log(Level.FINE, "Opened " + context.getConsumerGroupName());
   }
 
   @Override
-  public void onClose(PartitionContext context, CloseReason reason) throws Exception {
-    logger.log(Level.INFO,"Closed due to " + reason);
```

Comment:
logging at INFO level is not a great choice for routine stuff in libraries. In all zipkin components we use fine level.